### PR TITLE
Aligned the status code returned from Analyze to reflect the error

### PR DESCRIPTION
**Pull request by EranGon**

### CODE REVIEW

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs [hunk 1]
```diff
@@ -71,5 +74,34 @@ private static async Task<List<ContractsAlert>> RunSmartDetectorAsync(SmartDetec
...
+        /// <summary>
+        /// Converts an exception that was thrown from running a Smart Detector to the process's exit code
+        /// </summary>
+        /// <param name="e">The exception to convert</param>
+        /// <returns>The process's exit code</returns>
+        private static int ConvertExceptionToExitCode(Exception e)
+        {
+            switch (e)
+            {
+                case SmartDetectorRepositoryException _:
```
Comment (aviled) [inline src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: >RepositoryException [](start = 34, length = 19)

All repo exceptions are "not found"?
If we had an issue in the detector storage?

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs [hunk 2]
```diff
@@ -71,5 +74,34 @@ private static async Task<List<ContractsAlert>> RunSmartDetectorAsync(SmartDetec
...
+                    return (int)HttpStatusCode.NotFound;
+
+                case SmartDetectorLoadException _:
+                    return (int)HttpStatusCode.NotFound;
+
+                case IncompatibleResourceTypesException _:
+                    return (int)HttpStatusCode.BadRequest;
+
+                case AzureResourceManagerClientException armce:
+                    return (int)(armce.StatusCode ?? HttpStatusCode.BadRequest);
```
Comment (aviled) [inline src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: >?? [](start = 50, length = 2)

can you please add a comment explaining why if no status code found then bad request?
we don't want to retry on empty status code? how can this happen anyway?

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs (hunk 1 above)
Comment (EranGon) [reply src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: changed to SmartDetectorNotFoundException - but what should i do with SmartDetectorRepositoryException (which is a wrapper over StorageException)? return 500?

---
In reply to: [211210523](https://github.com/liadtal/Smart-Alerts/pull/204#discussion_r211210523) [](ancestors = 211210523)

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs [hunk 3]
```diff
@@ -71,5 +74,34 @@ private static async Task<List<ContractsAlert>> RunSmartDetectorAsync(SmartDetec
...
+
+                case UnidentifiedAlertResourceTypeException _:
+                    return (int)HttpStatusCode.BadRequest;
+
+                default:
+                    return (int)HttpStatusCode.BadRequest;
```
[the remark this answers is not in the document]
Comment (EranGon) [reply src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: my bad - changed to InternalServerError

---
In reply to: [211211346](https://github.com/liadtal/Smart-Alerts/pull/204#discussion_r211211346) [](ancestors = 211211346)

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs (hunk 2 above)
Comment (aviled) [reply src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: Try to find out when does this happen.
And if you can't then I would return a retriable status code

---
In reply to: [211241833](https://github.com/liadtal/Smart-Alerts/pull/204#discussion_r211241833) [](ancestors = 211241833,211210937)

Comment (EranGon) [reply src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: changed to return 500


---
In reply to: [211242476](https://github.com/liadtal/Smart-Alerts/pull/204#discussion_r211242476) [](ancestors = 211242476,211241833,211210937)

File: src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs (hunk 1 above)
Comment (aviled) [reply src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs]: why not get the storage exception status code?

---
In reply to: [211241915](https://github.com/liadtal/Smart-Alerts/pull/204#discussion_r211241915) [](ancestors = 211241915,211240680,211210523)

**Pull request merged**